### PR TITLE
Market: Test Publish with different providers

### PR DESCRIPTION
**Pull request by aarshkshah1992**
@anorth Tagging you as this is a small continuation of the previous PR for market tests.
This PR introduces tests for `PublishStorageDeals` with different providers.

### CODE REVIEW

File: actors/builtin/market/market_test.go [hunk 1]
```diff
@@ -404,6 +407,36 @@ func TestPublishStorageDeals(t *testing.T) {
 
 		totalStorageFee = big.Sum(totalStorageFee, deal4.TotalStorageFee(), deal5.TotalStorageFee())
 		require.EqualValues(t, totalStorageFee, st.TotalClientStorageFee)
+
+		// PUBLISH DEALS with a different provider
+		owner := tutil.NewIDAddr(t, 108)
+		provider := tutil.NewIDAddr(t, 109)
+		worker := tutil.NewIDAddr(t, 110)
```
Comment (anorth) [inline actors/builtin/market/market_test.go:414]: Only the provider address need to differ, so the test is slightly stronger if limited to that. Yes, it's realistic for providers to share owner/worker.

File: actors/builtin/market/market_test.go [hunk 2]
```diff
@@ -404,6 +407,36 @@ func TestPublishStorageDeals(t *testing.T) {
...
+
+		// generate second deal for second provider
+		deal7 := actor.generateDealAndAddFunds(rt, client1, miner, abi.ChainEpoch(25), abi.ChainEpoch(60))
+
+		// publish both the deals for the second provider
+		actor.publishDeals(rt, miner, deal6, deal7)
+
+		// assertions
+		rt.GetState(&st)
+		provider2Locked := big.Add(deal6.ProviderCollateral, deal7.ProviderCollateral)
```
Comment (anorth) [inline actors/builtin/market/market_test.go:428]: Check provider1Locked again too

File: actors/builtin/market/market_test.go (hunk 1 above)
Comment (aarshkshah1992) [reply actors/builtin/market/market_test.go]: This is done.

File: actors/builtin/market/market_test.go (hunk 2 above)
Comment (aarshkshah1992) [reply actors/builtin/market/market_test.go:426]: This is done.

**Pull request merged**